packages/js/customer-effort-score/src/customer-effort-score.tsx

```diff
@@ -13,6 +13,15 @@ import { CustomerFeedbackModal } from './customer-feedback-modal';
 
 const noop = () => {};
 
+type CustomerEffortScoreProps = {
+	recordScoreCallback: ( score: number, comments: string ) => void;
+	label: string;
+	onNoticeShownCallback?: () => void;
+	onNoticeDismissedCallback?: () => void;
+	onModalShownCallback?: () => void;
+	icon?: React.ReactNode;
```

Comment:
Because `useDispatch( 'core/notices2' )` isn't strictly typed, I'm wondering what happens if we call this without passing in icon?

Reply:
It's optional. The default value is null.
https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/notices/src/store/actions.js#L61

I've updated the type to `icon?: null | JSX.Element;` in e16aae9a82f3c25bec7a66db350f5753c1d77937 though.

Reply:
Ok great! Thanks for looking into this
That'll be a fun one to convert to Typescript... But that's a problem for another day 😆

Reply:
Ok great! Thanks for looking into this
That'll be a fun one to convert to Typescript... But that's a problem for another day 😆